hadoop-ozone/ozonefs/src/main/java/org/apache/hadoop/fs/ozone/OzoneFileSystem.java

```diff
@@ -669,6 +676,12 @@ public Path getWorkingDirectory() {
     return workingDir;
   }
 
+  @Override
+  public Token<?> getDelegationToken(String renewer) throws IOException {
+    return securityEnabled? adapter.getDelegationToken(renewer) :
+        super.getDelegationToken(renewer);
```

Comment:
bq. "should fetch DT from both", that is handled by addDelegationTokens() in the TokenIssuer.
Here we follow the FileSystem contract to return om delegation token if ozone security is enabled otherwise null as speced in FileSystem#getDelegationToken().